woocommerce/woocommerce-ios
# Add/Edit Products: update Product name UI

Fixes #1419 

## Changes
- Implemented all the changes in ViewModel and Datasource to accept the new Product name (alias title)
- Implemented EntityListiner in `ProductDetailsViewController` to observe changes on a single product, and for updating the UI when something changed.

## Testing
1) Navigate to the Products tab
2) Select a product
3) Press the Edit button
4) Tap title
5) Edit the title, press done, and make sure it changed on the previous screen and also on the product detail
6) Press again the Edit button, tap title and delete the content. Save.
7) Make sure the cell in the previous screen was updated following the cell style.

## Screenshots
| Title cell  |  Title empty |  Title editor |
:-------------------------:|:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2019-11-19 at 12 17 35](https://user-images.githubusercontent.com/495617/69142713-a61e6180-0ac7-11ea-80a9-479fb33e6584.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2019-11-19 at 12 17 41](https://user-images.githubusercontent.com/495617/69142756-c3ebc680-0ac7-11ea-852e-d4fb719e4fcf.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2019-11-20 at 11 23 43](https://user-images.githubusercontent.com/495617/69231105-ca8e4280-0b88-11ea-96a2-3ecf29aad774.png)



Update release notes:
